uvicorn/protocols/websockets/websockets_impl.py

```diff
@@ -231,6 +232,13 @@ async def asgi_send(self, message):
                 )
                 self.initial_response = None
                 self.accepted_subprotocol = message.get("subprotocol")
+                if "headers" in message:
+                    self.extra_headers.extend(
```

Comment:
Is there a need to `extend` here? Can't we just make an attribution and on the `init` not send the empty list?

Reply:
We need something mutable. The `handshake` is invoked before we are receiving headers via `send`.
So if we set `.extra_headers` to the proper value, it's too late because the `handshake` method was called with `None` value.

There is also the possibility to set `.extra_headers` as callable and that will be called during the handshake. What do you think about that?

Reply:
@Kludex , can you look at the proposal above?

Reply:
I'm fine with the current one. Thanks for the clarifications!